internal/v4/archive.go

```diff
@@ -309,7 +321,12 @@ func (h *ReqHandler) addEntity(id *router.ResolvedURL, r io.ReadSeeker, blobName
 	if id.URL.Series == "bundle" {
 		b, err := charm.ReadBundleArchiveFromReader(readerAt, contentLength)
 		if err != nil {
-			return errgo.Notef(err, "cannot read bundle archive")
+			switch errgo.Cause(err) {
+			case zip.ErrFormat, zip.ErrAlgorithm, zip.ErrChecksum:
```

Comment:
Nice, but why do we need this? Is this to provide more precise error messages to clients? If so, could you please add a comment?